packages/client/window/src/windows/Popup.ts

```diff
@@ -46,6 +47,10 @@ async function createPopup(url: string, options: PopupWindowOptions): Promise<Wi
         throw new Error("Failed to open popup window");
     }
 
+    if (options.awaitToLoad === false) {
```

Comment:
I'm not sure about this, but have you tried moving the `const _window` assignment inside the Promise?
This looks like a race condition: the _window's `onload` event might be firing before the `Promise` gets the chance  to hook the Promise resolution.

Reply:
Hmm I'm 99.9% sure that isn't the case, the reason is that the whole code here is sync and the window loading is an async action. While there might be a remote case that an async action takes less than sync ones (that's why I'm not 100% sure), this doesn't apply IMO

Reply:
I'm aware of that, otherwise I wouldn't talk about race conditions.
I think we are both saying the same thing! I'm assuming that `_window` is not a promise, but a sync object that fires off async operations in the background. Is there any way to initialize the `_window` with the hooks as parameters? That way you make sure they always get called.

Reply:
I also don't understand how is typescript allowing you to `return _window` when that's clearly not a `Promise<Window>`. Unless `Window extends Promise`???

Reply:
Resolving this to allow merging